### PR TITLE
DISPATCH-500 - Fixed code to use qdr_field_t to pass around the name …

### PR DESCRIPTION
**Pull request by ganeshmurthy**
…and identity of management objects

### CODE REVIEW

**Comment by grs** [comment]
Is this for DISPATCH-502?

**Comment by ganeshmurthy** [comment]
I verified that the newly added tests (test_check_address_name, test_check_link_route_name, test_check_auto_link_name) fail if this fix is not there

**Comment by ganeshmurthy** [comment]
@grs, I think that this patch might fix DISPATCH-502 as well. I am taking a look. Thanks.

File: src/router_config.c [hunk 1]
```diff
@@ -280,7 +280,12 @@ qd_error_t qd_router_configure_address(qd_router_t *router, qd_entity_t *entity)
         qd_parsed_field_t   *in_body = qd_parse(iter);
         qd_field_iterator_free(iter);
 
-        qdr_manage_create(router->router_core, 0, QD_ROUTER_CONFIG_ADDRESS, 0, in_body, 0);
+        qd_field_iterator_t *name_iter = 0;
+
+        if (name)
+            name_iter = qd_field_iterator_string(name);
```
Comment (ted-ross) [inline src/router_config.c]: I think you are going to leak this iterator.  After the call to qdr_manage_create, you must free the name_iter.

**Pull request closed without merging**